pkgs/applications/networking/browsers/firefox/common.nix

```diff
@@ -295,19 +326,88 @@ buildStdenv.mkDerivation ({
   ++ lib.optional enableOfficialBranding "--enable-official-branding"
   ++ extraConfigureFlags;
 
-  postConfigure = ''
-    cd obj-*
+  buildInputs = [
+    bzip2
+    dbus
+    dbus-glib
+    file
+    fontconfig
+    freetype
+    glib
+    gnum4
+    gtk3
+    icu
+    libffi
+    libGL
+    libGLU
+    libevent
+    libjpeg
+    libpng
+    libstartup_notification
+    libvpx
+    libwebp
+    nasm
+    nspr
+    nss
+    pango
+    perl
+    xorg.libX11
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXft
+    xorg.libXi
+    xorg.libXrender
+    xorg.libXt
+    xorg.libXtst
+    xorg.pixman
+    xorg.xorgproto
+    zip
+    zlib
+  ]
+  ++ lib.optional  alsaSupport alsa-lib
+  ++ lib.optional  pulseaudioSupport libpulseaudio # only headers are needed
+  ++ lib.optional  gssSupport libkrb5
+  ++ lib.optionals waylandSupport [ libxkbcommon libdrm ]
+  ++ lib.optional  jemallocSupport jemalloc
+  ++ extraBuildInputs;
+
+  profilingPhase = lib.optionalString pgoSupport ''
+    # Package up Firefox for profiling
+    ./mach package
+
+    # Run profiling
+    HOME=$TMPDIR \
+      LLVM_PROFDATA=llvm-profdata \
+      JARLOG_FILE="$TMPDIR/jarlog" \
+      xvfb-run -s "-screen 0 1920x1080x24 -nolisten local" \
+      ./mach python ./build/pgo/profileserver.py
+
+    # Copy profiling data to a place we can easily reference
+    cp ./merged.profdata $TMPDIR/merged.profdata
+
+    # Clean build dir
+    ./mach clobber
   '';
 
-  makeFlags = extraMakeFlags;
+  preBuild = ''
+    cd mozobj
+  '';
+
+  postBuild = ''
+    cd ..
+  '';
 
+  makeFlags = extraMakeFlags;
+  #separateDebugInfo = enableDebugSymbols;
```

Comment:
Had to disable `separateDebugInfo` because it would fail to create the debug output at the very end of the build process and I would love if you could have a look. @oxalica

Reply:
Do you have the build log? Building `firefox` is quite wild for my PC. :crying_cat_face:

Reply:
Of course with the latest changes not it works again. I assume overriding phases broke it. Thank you!